cmd/tetragon/main_test.go

```diff
@@ -52,15 +51,39 @@ func TestGeneratedExecEvents(t *testing.T) {
 	}
 	option.Config.ExportFilename = fname
 
-	err = tetragonExecuteCtx(ctx, cancel, ready)
-	assert.NoError(t, err)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ready := func() {
+		wg.Done()
+	}
+
+	// Start tetragon in separate process so we can keep the whole
+	// export/server machinery running until we get expected results.
+	go func() {
+		err = tetragonExecuteCtx(ctx, cancel, ready)
+		assert.NoError(t, err)
+	}()
+
+	// Wait till tetragon's observer is up and running
+	wg.Wait()
 
 	// Make sure exec event with pid 1 was generated
 	checker := ec.NewUnorderedEventChecker(
 		ec.NewProcessExecChecker("").WithProcess(
 			ec.NewProcessChecker().WithPid(1)),
 	)
 
-	err = jsonchecker.JsonTestCheck(t, checker)
+	// Try it 5 times and make sure exporter is up and processed all data
+	// in case it lags for some reason like slow CI server.
+	cnt := 0
+	for cnt < 5 {
+		if err = jsonchecker.JsonTestCheck(t, checker); err == nil {
```

Comment:
```suggestion
		if jsonchecker.JsonTestCheck(t, checker) == nil {
```
FYI the error you will check on line 88 will not be this one as this one is scoped only for this block.

Reply:
@mtardy hum, err is created on line 43, line 80 won't create new scope ('=' vs ':=' ) .. what do I miss?

Reply:
ah sorry I missed it somehow I read `:=` because I always use that in if statement... you are right!